### PR TITLE
CrdtMerge API web service

### LINKED ISSUE #1091: Consider design for CrdtMerge based on LfMerge
We want to copy some of LfMerge's design (and, probably, code) to use in a CrdtMerge program, to mediate between Crdt projects and the Send/Receive system.

I'll record notes in this issue, which we will eventually turn into a design document.

### PR DESCRIPTION
**Pull request by rmunn**
Fixes #1091.

Work done:

* Sync CRDT and FW projects
* Do Send/Receive with Lexbox after sync
* Sync CRDT and FW projects again if S/R pulled down any changes from others
* Create API endpoint to trigger all of this

Still remaining:

* <del>Create k8s deployment</del> Put off for later PR
* <del>Write tests to exercise it by sending API requests and verifying correct results happen</del> Put off for later PR

## To test

* Edit appsettings.Development.json and choose a different folder than `../../hgweb/repos` if you want, otherwise the test will put files into `lexbox/hgweb/repos`
* Run `dotnet run` inside the CrdtMerge folder
* Send a POST request to http://localhost:5275/sync?projectCode=sena-3
* Should take a minute or so to complete, watch the `dotnet run` logs while you wait
* Verify that the JSON result has the right number of changes (should be 1462, one per entry)
* Verify that the CRDT SQLite files showed up in `lexbox/hgweb/repos`, or wherever you put them

### CODE REVIEW

**Comment by rmunn** [comment]
Running into https://github.com/sillsdev/flexbridge/issues/409 while working on this. There's no reason built into FLExBridge why FixFwData.exe has to have an .exe filename. I created this problem for myself because I didn't fully understand how FlexBridge worked at the time, then spent a lot of effort working around the problem I myself created. But really, I should just relax the requirements of where FixFwData.exe needs to be located, and then everything will work as it should.

**Comment by rmunn** [comment]
About to rebase on top of current develop to fix LexBox.sln merge conflicts, then I'll take this out of draft status since it's ready for initial review. Not ready to push yet since I still need to create the k8s deployment (and figure out where it should store CRDT projects), but I just ran this successfully on my dev machine, and synced sena-3 from my local LexBox into a CRDT project.

**Comment by rmunn** [comment]
Now that this PR is ready, I should work on https://github.com/sillsdev/languageforge-lexbox/issues/1155 because I realized partway through that it's going to be necessary once we're doing S/R of real projects.

**Comment by rmunn** [comment]
A bunch of FwLite tests are now failing because of the changes we made to how projects are located. Will look at that tomorrow.

**Review by hahn-kev — DISMISSED** [review]

File: backend/CrdtMerge/SendReceiveHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        public string FwDataFile { get; } = Path.Join(Dir, $"{Code}.fwdata");
+    }
+
+    public record SendReceiveAuth(string Username, string Password)
+    {
+        public SendReceiveAuth(CrdtMergeConfig config) : this(config.LexboxUsername, config.LexboxPassword) { }
+    };
+
+    public record SendReceiveParams(string ProjectCode, string BaseUrl, string Dir) : ProjectPath(ProjectCode, Dir);
```
[the remark this answers is not in the document]
Comment (rmunn) [reply backend/CrdtMerge/SendReceiveHelpers.cs:18]: Yes, that's a copy-and-paste artifact that I missed.

**Review by hahn-kev — APPROVED** [review]

**Pull request merged**